semtypes/src/main/java/io/ballerina/types/Env.java

```diff
@@ -43,8 +52,29 @@ public Env() {
         this.recListAtoms = new ArrayList<>();
         recListAtoms.add(LIST_ATOMIC_RO);
         this.recMappingAtoms = new ArrayList<>();
+        recMappingAtoms.add(MAPPING_ATOMIC_RO);
         this.recFunctionAtoms = new ArrayList<>();
         types = new LinkedHashMap<>();
+
+        // Reserving the first two indexes of atomTable to represent cell VAL and cell NEVER typeAtoms.
```

Comment:
I think it is good if we can have a unit test somewhere that ensure `Environment` always get initialized correctly. We can use reflection to access the private states of the class without making them private. WDYT?

Reply:
+1. Test class added.